amd/src/script.js

```diff
@@ -1202,6 +1208,90 @@ define(['jquery', 'core/modal_factory', 'core/modal_events'], function($, ModalF
+                    basketButton.append(basket);
+
+                    var dropAreaText = document.createElement('p');
+                    dropAreaText.setAttribute('class', 'font-weight-bold text-white');
+                    dropAreaText.innerText = str('drop_here');
+
+                    var dropArea = document.createElement('div');
+                    dropArea.setAttribute('class',
+                        'h-100 w-100 position-absolute d-flex justify-content-center align-items-center');
+                    dropArea.append(dropAreaText);
+
+                    sections.forEach(section => {
+                        drag_event_listeners(section);
+                    });
+
+                    activities.forEach(activity => {
+                        drag_event_listeners(activity);
+                    });
+
+                    /**
+                     *  Initialize events for dragging
+                     * @param {object} draggable
+                     */
+                    function drag_event_listeners(draggable) {
+                        draggable.addEventListener('dragstart', (e) => {
+                            footerIconContainer.prepend(basketButton);
+                            sharingCartBlock.children[0].classList.add('dragging_item');
+                            sharingCartBlock.append(dropArea);
+                            currentDragging = e.target;
+                        });
+
+                        draggable.addEventListener('dragend', () => {
+                            footerIconContainer.removeChild(basketButton);
+                            sharingCartBlock.children[0].classList.remove('dragging_item');
+                            sharingCartBlock.removeChild(dropArea);
+                        });
+                    }
+
+                    [basketButton, sharingCartBlock].forEach((dropzone) => {
+                        dropzone.addEventListener("dragover", (e) => {
+                            e.preventDefault();
+                        });
+                        dropzone.addEventListener("dragenter", (e) => {
+                            e.preventDefault();
+                        });
+
+                        dropzone.addEventListener("drop", () => {
+                            if (typeof currentDragging === 'object') {
+                                currentDragging.querySelector('.add-to-sharing-cart').click();
+                            }
+
+                            currentDragging = undefined;
+                        });
+                    });
+                }
+
+                /**
+                 *  Make the first section (General) draggable
+                 */
+                function add_draggable_to_first_section() {
+                    document.getElementsByClassName("course-section-header")[0].classList.add('draggable');
+                    document.getElementsByClassName("course-section-header")[0].setAttribute('draggable', true);
```

Comment:
Is there a certainty that ``document.getElementsByClassName("course-section-header")`` will always have one element in the collection?
You can also reuse ``document.getElementsByClassName("course-section-header")`` as a variable instead of re-call the same thing twice.
```suggestion
                    const courseSectionHeader = document.getElementsByClassName("course-section-header")[0];
                    courseSectionHeader.classList.add('draggable');
                    courseSectionHeader.setAttribute('draggable', true);
```

Reply:
Changes added